homeassistant/components/growatt_server/sensor.py

```diff
@@ -45,8 +46,7 @@ def get_device_list(api, config):
         not login_response["success"]
         and login_response["msg"] == LOGIN_INVALID_AUTH_CODE
     ):
-        _LOGGER.error("Username, Password or URL may be incorrect!")
-        return
+        raise ConfigEntryAuthFailed("Username, Password or URL may be incorrect!")
```

Comment:
To be able to raise ConfigEntryAuthFailed you need to have a reauth flow and `growatt_server` does not have that one yet, so it should still use the logger instead

Reply:
I want to raise an exception here since this code should not return anything, otherwise it will fail at a more cryptic point later. Is there a more generic exception type I can use?

Reply:
Looking at similar situations in other integrations, `HomeAssistantError` seems appropiate.

Reply:
@joostlek Could you spare two minutes to confirm my change so perhaps it can be approved and merged? 👼